jupyter_ai_personas/pr_review_persona/persona.py

```diff
@@ -0,0 +1,232 @@
+                "Chat history: " + system_prompt
+            ],
+            markdown=True,
+            show_members_responses=True,
+            enable_agentic_context=True,
+            add_datetime_to_instructions=True,
+            tools=[
+                GithubTools( create_pull_request_comment= True, get_pull_requests= True, get_pull_request_changes= True),
+                ReasoningTools(add_instructions=True, think=True, analyze=True)
+            ]
+        )
+
+        return pr_review_team
+
+    async def process_message(self, message: Message):
+        provider_name = self.config.lm_provider.name
+        model_id = self.config.lm_provider_params["model_id"]
+
+        history = YChatHistory(ychat=self.ychat, k=2)
+        messages = await history.aget_messages()
+        
+        history_text = ""
+        if messages:
+            history_text = "\nPrevious conversation:\n"
+            for msg in messages:
+                role = "User" if isinstance(msg, HumanMessage) else "Assistant"
+                history_text += f"{role}: {msg.content}\n"
+
+        variables = PRPersonaVariables(
+            input=message.body,
+            model_id=model_id,
+            provider_name=provider_name,
+            persona_name=self.name,
+            context=history_text
+        )
+        
+        system_prompt = PR_PROMPT_TEMPLATE.format_messages(**variables.model_dump())[0].content
+        # team = self.initialize_team(system_prompt)
+        
+        try:
+            team = self.initialize_team(system_prompt)
+            response = team.run(message.body, 
+                              stream=False,
+                              stream_intermediate_steps=True,
+                              show_full_reasoning=True)
+
+            response = response.content
+            async def response_iterator():
+                yield response
+            
+            await self.stream_message(response_iterator())
+            
+        except ValueError as e:
+            error_message = f"Configuration Error: {str(e)}\nThis may be due to missing or invalid environment variables, model configuration, or input parameters."
+            async def error_iterator():
+                yield error_message
+            await self.stream_message(error_iterator())
+            
+        except boto3.exceptions.Boto3Error as e:
+            error_message = f"AWS Connection Error: {str(e)}\nThis may be due to invalid AWS credentials or network connectivity issues."
+            async def error_iterator():
+                yield error_message
+            await self.stream_message(error_iterator())
+            
+        except Exception as e:
+            error_message = f"PR Review Error ({type(e).__name__}): {str(e)}\nAn unexpected error occurred while the PR review team was analyzing your request."
+            async def error_iterator():
+                yield error_message
+            await self.stream_message(error_iterator())
```

Comment:
Doesn't seem like we have any content to stream here, can we just use `self.send_message` rather than creating iterators?